clang/lib/Sema/SemaType.cpp

```diff
@@ -1527,13 +1527,9 @@ static QualType ConvertDeclSpecToType(TypeProcessingState &state) {
     break;
   case DeclSpec::TST_half:    Result = Context.HalfTy; break;
   case DeclSpec::TST_BFloat16:
-    // Disable errors for SYCL and OpenMP device since definition of __bf16 is
-    // being moved to a shared header and it causes new errors emitted when
-    // host code is compiled with device compiler for SPIR target.
-    // FIXME: device code specific diagnostic is probably needed.
-    if (!S.Context.getTargetInfo().hasBFloat16Type() &&
```

Comment:
This and the following code was inadvertently added in a previous commit. This is not required anymore. Code here now matches community clang

Reply:
What previous commit was this?

Reply:
I think it was added with LLVM and SPIRV-LLVM-Translator pulldown (WW44)

Reply:
It was added here https://github.com/intel/llvm/pull/7186/commits/79509df3e233a522a72932070e1e2fccec53041e

Reply:
Thanks @Fznamznon.  

So if I understand correctly, it was not inadvertently added, but rather it was specifically added to suppress the new error messages seen.   We are now reverting that and the correct fix is to add the conditionalization instead in the headers.

Reply:
OpenMP code was added inadvertently. It is not needed here.
But for SYCL the previous fix wasn't enough. There was more diagnostics showing up after recent community changes in headers. It seemed better to guard header changes instead of playing "whack a mole" with more errors to suppress.

I think ideally in the long term we shouldn't modify the headers too. But it is not to easy to fix. We need to fix massive regression caused by the previous pulldown, so we asked @elizabethandrews to prepare this fix to unblock people.

I did some preliminary investigation regarding that, since I've noticed that some typedefs are actually under SSE macro. It seems if there is x86 aux triple is defined, we get all defined x86 macros while compiling for spir target. That seems like a bug in community code - https://github.com/llvm/llvm-project/blob/975740bf8db92e054b95a7cedd646d7bc3f74fed/clang/lib/Frontend/InitPreprocessor.cpp#L1332 .
If you look git blame around the code that I pointed, the community already tried to resolve this problem but failed.

Reply:
Thanks for the explanation.